src/git/repository.rs

```diff
@@ -1799,6 +1799,16 @@ pub fn find_repository(global_args: &Vec<String>) -> Result<Repository, GitAiErr
         )));
     }
 
+    // Rewrite global_args if -C path doesn't match the actual workdir/
+    // So every git command can assume it's being run in repo root
+    let mut global_args = global_args.clone();
+    if global_args.len() == 2 && global_args[0] == "-C" {
+        let workdir_str = workdir.display().to_string();
+        if global_args[1] != workdir_str {
+            global_args[1] = workdir_str;
+        }
+    }
```

Comment:
this doesn't change invocation arguments. It's only affecting the behavior of our execs to git and hardening the assumption they're running in root